src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -205,8 +204,16 @@ def _create_task_list(
     program_threads: number of threads to use per task
     For instance, if "threads" is set to 128 and "program_threads" to 32, then each task will run 4 batches in parallel.
 
-    We have chosen to use "taskfarmer" for parallelization, which means that "threads" and "program_threads" should
-    have the same value. This ensures that parallelization only happens between tasks, and not within them.
+    For tools capable of processing a batch of genomes, such as GTDB-TK and checkm2,
+    we have chosen to utilize the THREADS variable within Taskfarmer batch script (submit_taskfarmer.sl) for parallelization,
+    which means that "threads" and "program_threads" should have the same value.
+    This ensures that parallelization only happens between tasks, and not within them.
```

Comment:
This sentence is confusing to me - we're running GTDB_tk with multiple threads aren't we?

Reply:
yea. we run GTDB_TK with num of `program_threads`.

Reply:
This sentence seems to contradict that

Reply:
👍 how about now?

Reply:
I'm still confused I'm afraid. I think the problem is it's too easy to confuse "threads" because it could mean NODE_THREADS (e.g.. jobs) or os threads (e.g. _threads and _program_threads). Also, the number of threads is not necessarily correlated with the number of CPUs.

Also 
> This ensures that parallelization only happens between tasks, and not within them.

...but given _threads / _program_threads > 1 parallelization *is* happening within the task?

Reply:
yea. if _threads / _program_threads > 1 within task, there will be <_threads / _program_threads> number of parallelizations. That was actually my original design that we use `_threads` and `_program_threads` to control parallelization. 

https://github.com/kbase/collections/blob/develop/src/loaders/compute_tools/tool_common.py#L334

Reply:
Sorry, by `/` I meant "given _threads and _program_threads are both set to the same number greater than 1", which obviously is not how anyone reasonable is going to interpret `/`

Reply:
I made a comment below with edits with how I *think* things work, although I'm not sure I got things right

Reply:
👍 Thanks. I think I will start fixing it next anyways.

Reply:
There's also a TODO in there that needs addressing... and some grammar and spelling mistakes I just noticed, sigh

Reply:
👍 fixed grammar and also add explanation of `TODO: how does program_threads affect single genome tools?`.